pandas/core/frame.py

```diff
@@ -866,6 +866,12 @@ def dot(self, other):
         else:  # pragma: no cover
             raise TypeError('unsupported type: %s' % type(other))
 
+    def __matmul__(self, other):
+        dot_types = (Index, DataFrame, Series, np.ndarray, list)
+        if not isinstance(other, dot_types):
+            return NotImplemented
+        return self.dot(other)
```

Comment:
you don't need to specify the dot_types here, these are already handled in .dot (though maybe not tested fully). if something is rejected it should be there